rct229/ruleset_functions/normalize_space_schedules.py

```diff
@@ -0,0 +1,91 @@
+    "space": {
+        "$": ["lighting_space_type", "floor_area"],
+        "interior_lighting[*]": ["lighting_multiplier_schedule"],
+    }
+}
+
+
+def normalize_space_schedules(space, zone_height, schedules):
+    """This function would determine a normalized schedule for a data element in space.
+    NOTE: The function currently only works for interior lighting
+    Parameters
+    ----------
+    space: JSON - The space element that needs to determine a normalized schedule, e.g. space.interior_lighting
+    zone_height: float, average height of the zone
+    schedules: List[JSON] - schedule element
+    Returns
+    -------
+    A list containing 8760/8784 hourly values of a noralized schedule of the space data element
+    """
+    assert_required_fields(
+        GET_NORMALIZE_SPACE_SCHEDULE__REQUIRED_FIELDS["space"], space
+    )
+
+    space_total_power_per_area = 0.0
+    space_total_hourly_use_per_area_array = []
+
+    for interior_lighting in space.get("interior_lighting", []):
+        power_per_area = interior_lighting.get(
+            "power_per_area", 0.0
+        )  # ZERO POWER PER AREA / Power density
+        space_total_power_per_area += power_per_area
+
+        control_credit = 0.0
+        if interior_lighting.get(
+            "are_schedules_used_for_modeling_occupancy_control", None
+        ):
+            control_credit = table_G3_7_lookup(
+                lighting_space_type=getattr_(space, "space", "lighting_space_type"),
+                # occupancy control type can be None - simply ignored the credit
+                occupancy_control_type=interior_lighting.get(
+                    "occupancy_control_type", None
+                ),
+                space_height=zone_height,
+                space_area=getattr_(space, "space", "floor_area"),
+            )["control_credit"]
+
+        schedule_hourly_value = getattr_(
+            find_one_with_field_value(
+                "$", "id", interior_lighting["lighting_multiplier_schedule"], schedules
+            ),
+            "schedule",
+            "hourly_values",
+        )
+        hourly_use_per_area_array = [
+            hourly_value / (1.0 - control_credit) * power_per_area
+            for hourly_value in schedule_hourly_value
+        ]
+
+        if not space_total_hourly_use_per_area_array:
+            # if the space_total_hourly_use_per_area_array is empty
+            space_total_hourly_use_per_area_array = [0.0] * len(
+                hourly_use_per_area_array
+            )
+
+        # make sure the hourly schedules have the same number of hours
+        assert_(
+            len(space_total_hourly_use_per_area_array)
+            == len(hourly_use_per_area_array),
+            f"Number of hours mismatched. The space has number of hours of: {len(space_total_hourly_use_per_area_array)} "
+            f"but the schedule id {interior_lighting['lighting_multiplier_schedule']} has number of hours of: "
```

Comment:
Is this proper Python syntax to concatenate these strings?  I think you need to put a `+` between them.

Reply:
I think this is black or isort did it...
I am not sure what is more pythonic way to do it..